UefiPayloadPkg/Library/BuildFdtLib/X86_BuildFdtLib.c

```diff
@@ -0,0 +1,978 @@
+                                              PCI_SVID_OFFSET,
+                                              sizeof (PciData.Device.SubsystemVendorID),
+                                              &(PciData.Device.SubsystemVendorID)
+                                              );
+
+          Status = mPciDevicePpi->PciIo.Pci.Read (
+                                              &mPciDevicePpi->PciIo,
+                                              (EFI_PCI_IO_PROTOCOL_WIDTH)EfiPciWidthUint16,
+                                              PCI_SID_OFFSET,
+                                              sizeof (PciData.Device.SubsystemID),
+                                              &(PciData.Device.SubsystemID)
+                                              );
+        }
+
+        Data16 = CpuToFdt16 (PciData.Device.SubsystemID);
+        Status = FdtSetProp (Fdt, GmaNode, "subsystem-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Device.SubsystemVendorID);
+        Status = FdtSetProp (Fdt, GmaNode, "subsystem-vendor-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Hdr.RevisionID);
+        Status = FdtSetProp (Fdt, GmaNode, "revision-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Hdr.DeviceId);
+        Status = FdtSetProp (Fdt, GmaNode, "device-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Hdr.VendorId);
+        Status = FdtSetProp (Fdt, GmaNode, "vendor-id", &Data16, sizeof (UINT16));
+      }
+
+      if (SerialParent != NULL) {
+        DEBUG ((DEBUG_INFO, "SerialParent->IsIsaCompatible  :%x , SerialParent->ParentDevicePcieBaseAddress :%x\n", SerialParent->IsIsaCompatible, SerialParent->ParentDevicePcieBaseAddress));
+      }
+
+      {
+        eSPINode = TempNode;
+        if (SerialParent->IsIsaCompatible && (SerialParent != NULL)) {
+          BusBase  = (SerialParent->ParentDevicePcieBaseAddress >> 20) & 0xFF;
+          DevBase  = (SerialParent->ParentDevicePcieBaseAddress >> 15) & 0x1F;
+          FunBase  = (SerialParent->ParentDevicePcieBaseAddress >> 12) & 0x07;
+          Status   = AsciiSPrint (eSPIStr, sizeof (eSPIStr), "isa@%X,%X", DevBase, FunBase);
+          eSPINode = FdtAddSubnode (Fdt, TempNode, eSPIStr);
+          Status   = FdtSetProp (Fdt, eSPINode, "compatible", "isa", (UINT32)(AsciiStrLen ("isa")+1));
+          ASSERT_EFI_ERROR (Status);
+          Data16 = CpuToFdt16 (1);
+          Status = FdtSetProp (Fdt, eSPINode, "#size-cells", &Data16, sizeof (UINT16));
+          Data16 = CpuToFdt16 (2);
+          Status = FdtSetProp (Fdt, eSPINode, "#address-cells", &Data16, sizeof (UINT16));
+        }
+
+        DEBUG ((DEBUG_INFO, "BusBase  :%x , PciRootBridgeInfo->RootBridge[Index].Bus.Base :%x\n", BusBase, PciRootBridgeInfo->RootBridge[Index].Bus.Base));
+        if (BusBase == PciRootBridgeInfo->RootBridge[Index].Bus.Base) {

```

Comment:
I think we should compare BusBase vs Bus.Base in above line#738 and only create isa node when BusBase was "within" current pci-rb range. (otherwise we should create isa in another pci-rb later)

Reply:
resolved